src/lib/index.jsx

```diff
@@ -250,7 +250,27 @@ class OtpInput extends Component {
 
   // The content may not have changed, but some input took place hence change the focus
   handleOnInput = (e) => {
-    if (this.isInputValueValid(e.target.value)) {
+    if (!e.target.value) return;
+    if (e.target.value && e.target.value.length > 1) {
+      // this code happens when the users click the 'from your messages' code at the top of the keyboard
+      // essentially the mobile OS's act as though someone has pressed all of the OTP numbers all at once
+      // rather than using a 'paste' style.
+      e.preventDefault();
+      const { numInputs } = this.props;
+      const { activeInput } = this.state;
+      const otp = this.getOtpValue();
+
+      // Get pastedData in an array of max size (num of inputs - current position)
+      const pastedData = e.target.value.slice(0, numInputs - activeInput).split('');
+      // Paste data from focused input onwards
+      for (let pos = 0; pos < numInputs; ++pos) {
+        if (pos >= activeInput && pastedData.length > 0) {
+          otp[pos] = pastedData.shift();
+        }
+      }
+      this.handleOtpChange(otp);
+      this.focusInput(this.props.numInputs);
```

Comment:
Should this actually be...?
```suggestion
      this.focusInput(otp.length);
```

(I think this would only make a difference if the OTP suggested by the OS is shorter than the specified `numInputs`)